llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUInstPrinter.cpp

```diff
@@ -813,7 +813,11 @@ void AMDGPUInstPrinter::printRegularOperand(const MCInst *MI, unsigned OpNo,
           OpInfo, STI.getHwMode(MCSubtargetInfo::HwMode_RegInfo));
       const MCRegisterClass &RC = MRI.getRegClass(RCID);
       auto Reg = mc2PseudoReg(Op.getReg());
-      if (!RC.contains(Reg) && !isInlineValue(Reg)) {
+      bool Err = !RC.contains(Reg) && !isInlineValue(Reg);
+      if (!Err && (RCID == AMDGPU::VGPR_32_Lo128RegClassID ||
+                   RCID == AMDGPU::VS_32_Lo128RegClassID))
+        Err = Reg >= AMDGPU::VGPR128 && Reg <= AMDGPU::VGPR1023;
```

Comment:
Not a big deal, but should we somehow factor this logic?
I feel it is easy to miss the RCID (`RCID == AMDGPU::VGPR_32_Lo128RegClassID || RCID == AMDGPU::VS_32_Lo128RegClassID`) and/or the range (`Reg >= AMDGPU::VGPR128 && Reg <= AMDGPU::VGPR1023;`).
And we're repeating it twice in this patch (the second time in `AMDGPUOperand::isRegClass`).

I'm afraid finding all the places where this needs to be fixed to be problematic if it's not shared.